app/knight.py

```diff
@@ -0,0 +1,29 @@
+class Knight:
+
+    def __init__(self, name: str, power: int, hp: int,
+                 armour: list, weapon: dict, potion: dict):
+
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+
+    def apply_armour(self):
+        self.protection = sum(item["protection"] for item in self.armour)
+        return self
+
+    def apply_weapon(self):
+        self.power += self.weapon["power"]
+        return self
+
+    def apply_potion_if_any(self):
+        if self.potion is not None:
+            if "power" in self.potion["effect"]:
+                self.power += self.potion["effect"]["power"]
+            if "hp" in self.potion["effect"]:
+                self.hp += self.potion["effect"]["hp"]
+            if "protection" in self.potion["effect"]:
+                self.protection += self.potion["effect"]["protection"]
```

Comment:
These code duplications can be also fixed, using stats = ("protection", "power", "hp") and looping on stats

Reply:
don't really understand how to use this, can you give a clue?

Following code will not work because of self.stat:
_for stat in ("power", "hp", "protection"):
    if stat in self.potion["effect"]:
        self.stat += self.potion["effect"][stat]_

Reply:
```
stats = ("protection", "power", "hp")

if knight["potion"] is not None:

                for statistic in stats:
                    if statistic in knight["potion"]["effect"]:
                        knight[statistic] += knight["potion"]["effect"][
                            statistic
                        ]
```

Reply:
Hmm..
I didn’t notice self.
We’ll probably have to leave it at that.